tests/acceptance/test_hooked_transformer.py

```diff
@@ -167,6 +167,66 @@ def test_from_pretrained_revision():
         raise AssertionError("Should have raised an error")
 
 
+def check_norm_folding(
+    model_name,
+    hf_model=None,
+    tokenizer=None,
+    prompt="Hello, world!",
+    device=None,
+    dtype=None,
+):
+    """
+    Checks that loading a model with Layer/RMS Norm folding enabled does not (significantly) change its outputs.
+
+    :return: The maximum difference between the logits produced by the same model with and without norm folding enabled.
+    """
+
+    # If a device/dtype is not specified, and hf_model is provided, use its device/dtype
+    # Otherwise, default to cuda (if available)/float32
+    if device is None:
+        if hf_model:
+            device = hf_model.device
+        else:
+            device = "cuda" if torch.cuda.is_available() else "cpu"
+    if dtype is None:
+        if hf_model:
+            dtype = hf_model.dtype
+        else:
+            dtype = "float32"
+
+    folded = HookedTransformer.from_pretrained(
+        model_name=model_name,
+        hf_model=hf_model,
+        device=device,
+        tokenizer=tokenizer,
+        dtype=dtype,
+        fold_ln=True,
+    )
+    tokens = folded.tokenizer.encode(prompt, return_tensors="pt")
+    folded_logits = folded(tokens).detach()
+    del folded
+    torch.cuda.empty_cache()
+
+    unfolded = HookedTransformer.from_pretrained(
+        model_name=model_name,
+        hf_model=hf_model,
+        device=device,
+        tokenizer=tokenizer,
+        dtype=dtype,
+        fold_ln=False,
+    )
+    unfolded_logits = unfolded(tokens).detach()
+    del unfolded
+    torch.cuda.empty_cache()
+
+    return torch.max(
```

Comment:
Nit: Can you instead use torch.allclose? As in `check_similarity_with_hf_model`

Reply:
I went with the max diff instead of an allclose check since the tolerance varies greatly with the dtype used, float32 is <1e-5 while float16 is ~1e-2, so it seemed useful to be able to check how the tolerance is affected. I can switch it back/add an allclose assertion if this isn't needed though

Reply:
Fair enough, I'm happy for you to leave as is then.

Reply:
Can you add a comment explaining?

Reply:
Sure, I may also add the allclose check for some tolerance that passes even on fp16, but continue returning the tolerance